src/pages/ProfilesExplorerView/components/SceneExploreDiffFlameGraph/infrastructure/useFetchDiffProfile.ts

```diff
@@ -0,0 +1,91 @@
+import { TimeRange } from '@grafana/data';
+import { useMaxNodesFromUrl } from '@shared/domain/url-params/useMaxNodesFromUrl';
+import { queryClient } from '@shared/infrastructure/react-query/queryClient';
+import { useQuery } from '@tanstack/react-query';
+import { useEffect } from 'react';
+
+import { DataSourceProxyClientBuilder } from '../../../infrastructure/series/http/DataSourceProxyClientBuilder';
+import { DiffProfileApiClient } from './DiffProfileApiClient';
+
+type FetchParams = {
+  dataSourceUid: string;
+  serviceName: string;
+  baselineTimeRange: TimeRange;
+  baselineQuery: string;
+  comparisonTimeRange: TimeRange;
+  comparisonQuery: string;
+};
+
+export function useFetchDiffProfile({
+  dataSourceUid,
+  serviceName,
+  baselineTimeRange,
+  baselineQuery,
+  comparisonTimeRange,
+  comparisonQuery,
+}: FetchParams) {
+  const [maxNodes] = useMaxNodesFromUrl();
+
+  // we use "raw" to cache relative time ranges between renders, so that only refetch() will trigger a new query
+  const queryKey = [
+    'diff-profile',
+    baselineQuery,
+    baselineTimeRange?.raw.from.toString(),
+    baselineTimeRange?.raw.to.toString(),
+    comparisonQuery,
+    comparisonTimeRange?.raw.from.toString(),
+    comparisonTimeRange?.raw.to.toString(),
+    maxNodes,
+  ];
+
+  useEffect(() => {
```

Comment:
**suggestion:** Would be nice to have some comments why it's needed

**question:** Is it expected that it's called on the first run + every time a dependency changes?